CMakeLists.txt

```diff
@@ -91,8 +88,18 @@ target_include_directories(concurrencpp
 
 target_compile_features(concurrencpp PUBLIC cxx_std_20)
 
+set_target_properties(concurrencpp PROPERTIES
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN ON
+)
+
 target_coroutine_options(concurrencpp)
 
+target_compile_definitions(concurrencpp
+        PRIVATE $<$<STREQUAL:$<TARGET_PROPERTY:concurrencpp,TYPE>,SHARED_LIBRARY>:CRCPP_EXPORT_API>
+        INTERFACE $<$<STREQUAL:$<TARGET_PROPERTY:concurrencpp,TYPE>,SHARED_LIBRARY>:CRCPP_IMPORT_API>
+)
```

Comment:
Eh, I think whether to use a generator expression or if statements is a personal preference.

As for GenerateExportHeader, I used to use it, but because it produces different headers based on the compiler + os, I tend to just prefer a header that can be used portably instead nowadays.

Reply:
I think it'd be better to stick to the CMake built-in for now. If issues arise due to this later down the line, things can be changed around then.

Reply:
I'm not sure what you me by "CMake built-in". Do you mean to use GenerateExportHeader? If so, I strongly disagree. My environment requires multiple compilers (some libraries are built with the intel compiler), so having portable headers is very important to me.

Reply:
@friendlyanon @ohanar what is the status of it?

Reply:
@friendlyanon @ohanar what is the status of it?

Reply:
I find the logic in `platform_defs.h` unusual, but it works. I would still keep an eye on `__has_cpp_attribute(gnu::visibility)` regardless, mainly because of unfamiliarity.